srcCpp/perftest_publisher.cxx

```diff
@@ -1836,10 +1872,13 @@ int perftest_cpp::Publisher()
 
                     // flush anything that was previously sent
                     writer->Flush();
-                    writer->wait_for_acknowledgments(
-                            timeout_wait_for_ack_sec,
-                            timeout_wait_for_ack_nsec);
-                    announcement_reader_listener->announced_subscribers = _NumSubscribers;
+                    if (_IsReliable) {
+                        writer->wait_for_acknowledgments(
+                                timeout_wait_for_ack_sec,
+                                timeout_wait_for_ack_nsec);
+                    } else {
+                        MilliSleep(timeout_wait_for_ack_nsec / 1000000);
+                    }
```

Comment:
We still need to decide if we want to keep this approach or move the if(_isReliable) to the writer wait_for_acknowledgments call. This would require the writer to know about its reliability (which makes sense, but it is not propagated to the writer in the current implementation.

@fercsrti77

Reply:
@fercsrti77, I implemented this option but then I also implemented your suggestion of using the waitForAck function. I believe the second option makes more sense.